### PR TITLE
Implement diffable for Label component

### PR DESCRIPTION
**Pull request by pablog**
# Known issues:

Not taking care of font size. Questions:

- What to do if there are multiple lines in the label?
- `maximumFontSize` should take into account `text` and `textAfterLayout`?

Would like to do PP with this.

### CODE REVIEW

File: Portal/View/Components/Label.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 import Foundation
 
-public struct LabelProperties {
+public struct LabelProperties<MessageType>: AutoPropertyDiffable {
```
Comment (guidomb) [inline Portal/View/Components/Label.swift]: Why are we adding `MessageType`?

File: PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                    properties: labelProperties,
+                    styleSheet: labelStyle,
+                    layout: layout()
+                )
+                
+            }
+
+            context("when the change set contains label property changes") {
+
+                it("applies 'text' property changes") {
```
Comment (guidomb) [inline PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift]: This should be refactored to

```swift
let result = label.apply(changeSet: changeSet, layoutEngine: layoutEngine)
result.afterLayoutTask?()
expect(label.text).to(equal("Hello World"))
```

**Review by guidomb — CHANGES_REQUESTED** [review]

**Comment by guidomb** [comment]
I would suggest to add an example screen for labels exercising different use cases

- Label with single line text fixed text size
- Label with multiple line text fixed text size
- Label with singe line dynamic text size, we should check that if we put a text size bigger than the container then the text size gets reduced

Regarding your questions

> What to do if there are multiple lines in the label?

This should work as expected in UIKit 

> maximumFontSize should take into account text and textAfterLayout?

No

File: PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift (hunk 2 above)
Comment (pablog) [reply PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift]: Nice, that's what I wanted.

I thought `afterLayoutTask` was a closure executed after layout.

File: Portal/View/Component.swift [hunk 3]
```diff
@@ -137,7 +137,7 @@ extension Component {
             return .button(properties.map(transform), style, layout)
 
         case .label(let properties, let style, let layout):
-            return .label(properties, style, layout)
+            return .label(properties.map(transform), style, layout)
```
Comment (aducret) [inline Portal/View/Component.swift]: Why this change? All the components that has a generic `<MessageType>` should use this transform? @guidomb

File: Portal/View/Components/Label.swift [hunk 4]
```diff
@@ -78,3 +88,37 @@ public func labelStyleSheet(
     configure(&base, &component)
     return StyleSheet(component: component, base: base)
 }
+
+// MARK: - Change set
+
+internal struct LabelChangeSet<MessageType> {
```
Comment (aducret) [inline Portal/View/Components/Label.swift]: Remove the `MessageType` here too.

File: Portal/View/UIKit/Renderers/LabelRenderer.swift [hunk 5]
```diff
@@ -8,51 +8,103 @@
 
 import UIKit
 
+public let defaultLabelFontSize = UInt(UIFont.systemFontSize)
```
Comment (aducret) [inline Portal/View/UIKit/Renderers/LabelRenderer.swift]: There is a `UIFont.labelFontSize`, I think it would be better.

File: PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        beforeEach {
+            layoutEngine = YogaLayoutEngine()
+        }
+
+        describe(".apply(changeSet: TextFieldChangeSet) -> Result") {
+
+            var changeSet: LabelChangeSet<String>!
+
+            beforeEach {
+
```
Comment (aducret) [inline PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift]: Remove this empty line.

File: PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                    label.numberOfLines = 0
+                    label.minimumScaleFactor = 1.0
+                }
+
+                changeSet = LabelChangeSet.fullChangeSet(
+                    properties: labelProperties,
+                    styleSheet: labelStyle,
+                    layout: layout()
+                )
+
```
Comment (aducret) [inline PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift]: Remove this empty line.

File: PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift (hunk 2 above)
Comment (aducret) [reply PortalTests/View/UIKit/Renderers/LabelRendererSpec.swift]: Don't you should check `text` property? (before layout)

**Review by aducret — CHANGES_REQUESTED** [review]

File: Example/Views/ModalScreen.swift [hunk 8]
```diff
@@ -23,7 +23,28 @@ enum ModalScreen {
             root: .stack(ExampleApplication.navigationBar(title: "Modal")),
             component: container(
                 children: [
-                    label(text: "Modal screen"),
+                    label(
```
Comment (aducret) [inline Example/Views/ModalScreen.swift]: Maybe this changes in label could be in examples view

**Review by guidomb — APPROVED** [review]
